ui/app/ducks/metamask/metamask.js

```diff
@@ -304,64 +304,39 @@ export default function reduceMetamask (state = {}, action) {
         send: newSend,
       })
 
-      case actionConstants.UPDATE_SWAP_FROM_TOKEN:
-        const newSwapFrom = {
-          ...metamaskState.swap,
-          tokenFrom: action.value,
-        }
-        // erase token-related state when switching back to native currency
-        if (newSwapFrom.editingTransactionId && !newSwapFrom.token) {
-          const unapprovedTx = newSwapFrom?.unapprovedTxs?.[newSwapFrom.editingTransactionId] || {}
-          const txParams = unapprovedTx.txParams || {}
-          Object.assign(newSwapFrom, {
-            tokenBalance: null,
-            balance: '0',
-            from: unapprovedTx.from || '',
-            unapprovedTxs: {
-              ...newSwapFrom.unapprovedTxs,
-              [newSwapFrom.editingTransactionId]: {
-                ...unapprovedTx,
-                txParams: {
-                  ...txParams,
-                  data: '',
-                },
-              },
-            },
-          })
```

Comment:
**Note to reviewer:**

I don't think all this stuff is relevant for Swap. Moreover, `editingTransactionId` is always unset in our case. Please let me know if this is incorrect.

Reply:
This is fine!